### ISSUE TITLE
extension hot-reloading

### ISSUE THREAD

**Issue by boneskull** [appium/appium#18439]
While the `APPIUM_RELOAD_EXTENSIONS` env var, if present, will reload _only_ the entry point of an extension upon new session creation, this functionality should be built-out further:

1. It should reload the entire extension; not just the entry point
2. It may make sense to enable a reload before any command (not just new session), or alternatively, use a file watcher
3. We may want to enable this behavior _by default_ if we detect we are in "dev mode".

There are a few different ways to approach this, but would rather avoid adding a ton of code to make this work.  Can we pull in a lib to help?  Open to any and all ideas.

**Comment by jlipps** [comment]
I have no other ideas beyond my already too-naive current implementation 😆

**Comment by mykola-mokhnach** [comment]
I don't think we are going to work on that unless requested by multiple users. Closing for now
